recipe-generator/src/main/java/ca/derekcormier/recipe/generator/TypescriptFilters.java

```diff
@@ -0,0 +1,157 @@
+package ca.derekcormier.recipe.generator;
+
+import java.util.Map;
+
+import ca.derekcormier.recipe.cookbook.Cookbook;
+import ca.derekcormier.recipe.cookbook.CookbookUtils;
+import ca.derekcormier.recipe.cookbook.Primitive;
+import ca.derekcormier.recipe.cookbook.type.ArrayType;
+import ca.derekcormier.recipe.cookbook.type.EnumType;
+import ca.derekcormier.recipe.cookbook.type.FlagType;
+import ca.derekcormier.recipe.cookbook.type.ParamType;
+import ca.derekcormier.recipe.cookbook.type.PrimitiveType;
+import ca.derekcormier.recipe.cookbook.type.Type;
+import liqp.filters.Filter;
+
+public class TypescriptFilters {
+    public static Filter createTsTypeFilter(Cookbook cookbook) {
+        return new Filter("tstype") {
+            @Override
+            public Object apply(Object value, Object... params) {
+                ParamType type = CookbookUtils.parseType(super.asString(value), cookbook);
+                return toTsType(type, params.length > 0 && super.asBoolean(params[0]));
+            }
+
+            private String toTsType(ParamType type, boolean varargAsArray) {
+                String tsType = _toTsType(type.getType());
+                if (type.isVararg()) {
+                    if (varargAsArray) {
+                        tsType += "[]";
+                    }
+                    else {
+                        tsType += "...";
+                    }
+                }
+                return tsType;
+            }
+
+            private String _toTsType(Type type) {
+                if (type instanceof PrimitiveType) {
+                    switch (((PrimitiveType)type).getPrimitive()) {
+                        case BOOLEAN:
+                            return "boolean";
+                        case INTEGER:
+                            return "number";
+                        case FLOAT:
+                            return "number";
+                        case STRING:
+                            return "string";
+                        default:
+                            throw new RuntimeException("unknown type");
+                    }
+                }
+                else if (type instanceof FlagType) {
+                    return "boolean";
```

Comment:
a `FlagType` is a boolean?

Reply:
Internally, yeah. A `flag` type for optionals just means you can do `new MyIngredient.withX()`, which indicates that x was "flagged", rather than specifying boolean like `MyIngredient.withX(true)`.